### PR TITLE
Fix wasm32 builds

### PR DESCRIPTION
**Pull request by junderw**
This was causing problems when building.

<details><summary>Expand</summary>
<p>

```
error: linking with `rust-lld` failed: exit status: 1
  |
  = note: rust-lld: error: duplicate symbol: WASM32_INT_SIZE
          >>> defined in /tmp/rustciBlixx/libsecp256k1_sys-b664e02e63969bc9.rlib(precomputed_ecmult_gen.o)
          >>> defined in /tmp/rustciBlixx/libsecp256k1_sys-b664e02e63969bc9.rlib(precomputed_ecmult.o)
          
          rust-lld: error: duplicate symbol: WASM32_INT_ALIGN
          >>> defined in /tmp/rustciBlixx/libsecp256k1_sys-b664e02e63969bc9.rlib(precomputed_ecmult_gen.o)
          >>> defined in /tmp/rustciBlixx/libsecp256k1_sys-b664e02e63969bc9.rlib(precomputed_ecmult.o)
          
          rust-lld: error: duplicate symbol: WASM32_UNSIGNED_INT_SIZE
          >>> defined in /tmp/rustciBlixx/libsecp256k1_sys-b664e02e63969bc9.rlib(precomputed_ecmult_gen.o)
          >>> defined in /tmp/rustciBlixx/libsecp256k1_sys-b664e02e63969bc9.rlib(precomputed_ecmult.o)
          
          rust-lld: error: duplicate symbol: WASM32_UNSIGNED_INT_ALIGN
          >>> defined in /tmp/rustciBlixx/libsecp256k1_sys-b664e02e63969bc9.rlib(precomputed_ecmult_gen.o)
          >>> defined in /tmp/rustciBlixx/libsecp256k1_sys-b664e02e63969bc9.rlib(precomputed_ecmult.o)
          
          rust-lld: error: duplicate symbol: WASM32_SIZE_T_SIZE
          >>> defined in /tmp/rustciBlixx/libsecp256k1_sys-b664e02e63969bc9.rlib(precomputed_ecmult_gen.o)
          >>> defined in /tmp/rustciBlixx/libsecp256k1_sys-b664e02e63969bc9.rlib(precomputed_ecmult.o)
          
          rust-lld: error: duplicate symbol: WASM32_SIZE_T_ALIGN
          >>> defined in /tmp/rustciBlixx/libsecp256k1_sys-b664e02e63969bc9.rlib(precomputed_ecmult_gen.o)
          >>> defined in /tmp/rustciBlixx/libsecp256k1_sys-b664e02e63969bc9.rlib(precomputed_ecmult.o)
          
          rust-lld: error: duplicate symbol: WASM32_UNSIGNED_CHAR_SIZE
          >>> defined in /tmp/rustciBlixx/libsecp256k1_sys-b664e02e63969bc9.rlib(precomputed_ecmult_gen.o)
          >>> defined in /tmp/rustciBlixx/libsecp256k1_sys-b664e02e63969bc9.rlib(precomputed_ecmult.o)
          
          rust-lld: error: duplicate symbol: WASM32_UNSIGNED_CHAR_ALIGN
          >>> defined in /tmp/rustciBlixx/libsecp256k1_sys-b664e02e63969bc9.rlib(precomputed_ecmult_gen.o)
          >>> defined in /tmp/rustciBlixx/libsecp256k1_sys-b664e02e63969bc9.rlib(precomputed_ecmult.o)
          
          rust-lld: error: duplicate symbol: WASM32_PTR_SIZE
          >>> defined in /tmp/rustciBlixx/libsecp256k1_sys-b664e02e63969bc9.rlib(precomputed_ecmult_gen.o)
          >>> defined in /tmp/rustciBlixx/libsecp256k1_sys-b664e02e63969bc9.rlib(precomputed_ecmult.o)
          
          rust-lld: error: duplicate symbol: WASM32_PTR_ALIGN
          >>> defined in /tmp/rustciBlixx/libsecp256k1_sys-b664e02e63969bc9.rlib(precomputed_ecmult_gen.o)
          >>> defined in /tmp/rustciBlixx/libsecp256k1_sys-b664e02e63969bc9.rlib(precomputed_ecmult.o)
          
          rust-lld: error: duplicate symbol: WASM32_INT_SIZE
          >>> defined in /tmp/rustciBlixx/libsecp256k1_sys-b664e02e63969bc9.rlib(precomputed_ecmult.o)
          >>> defined in /tmp/rustciBlixx/libsecp256k1_sys-b664e02e63969bc9.rlib(secp256k1.o)
          
          rust-lld: error: duplicate symbol: WASM32_INT_ALIGN
          >>> defined in /tmp/rustciBlixx/libsecp256k1_sys-b664e02e63969bc9.rlib(precomputed_ecmult.o)
          >>> defined in /tmp/rustciBlixx/libsecp256k1_sys-b664e02e63969bc9.rlib(secp256k1.o)
          
          rust-lld: error: duplicate symbol: WASM32_UNSIGNED_INT_SIZE
          >>> defined in /tmp/rustciBlixx/libsecp256k1_sys-b664e02e63969bc9.rlib(precomputed_ecmult.o)
          >>> defined in /tmp/rustciBlixx/libsecp256k1_sys-b664e02e63969bc9.rlib(secp256k1.o)
          
          rust-lld: error: duplicate symbol: WASM32_UNSIGNED_INT_ALIGN
          >>> defined in /tmp/rustciBlixx/libsecp256k1_sys-b664e02e63969bc9.rlib(precomputed_ecmult.o)
          >>> defined in /tmp/rustciBlixx/libsecp256k1_sys-b664e02e63969bc9.rlib(secp256k1.o)
          
          rust-lld: error: duplicate symbol: WASM32_SIZE_T_SIZE
          >>> defined in /tmp/rustciBlixx/libsecp256k1_sys-b664e02e63969bc9.rlib(precomputed_ecmult.o)
          >>> defined in /tmp/rustciBlixx/libsecp256k1_sys-b664e02e63969bc9.rlib(secp256k1.o)
          
          rust-lld: error: duplicate symbol: WASM32_SIZE_T_ALIGN
          >>> defined in /tmp/rustciBlixx/libsecp256k1_sys-b664e02e63969bc9.rlib(precomputed_ecmult.o)
          >>> defined in /tmp/rustciBlixx/libsecp256k1_sys-b664e02e63969bc9.rlib(secp256k1.o)
          
          rust-lld: error: duplicate symbol: WASM32_UNSIGNED_CHAR_SIZE
          >>> defined in /tmp/rustciBlixx/libsecp256k1_sys-b664e02e63969bc9.rlib(precomputed_ecmult.o)
          >>> defined in /tmp/rustciBlixx/libsecp256k1_sys-b664e02e63969bc9.rlib(secp256k1.o)
          
          rust-lld: error: duplicate symbol: WASM32_UNSIGNED_CHAR_ALIGN
          >>> defined in /tmp/rustciBlixx/libsecp256k1_sys-b664e02e63969bc9.rlib(precomputed_ecmult.o)
          >>> defined in /tmp/rustciBlixx/libsecp256k1_sys-b664e02e63969bc9.rlib(secp256k1.o)
          
          rust-lld: error: duplicate symbol: WASM32_PTR_SIZE
          >>> defined in /tmp/rustciBlixx/libsecp256k1_sys-b664e02e63969bc9.rlib(precomputed_ecmult.o)
          >>> defined in /tmp/rustciBlixx/libsecp256k1_sys-b664e02e63969bc9.rlib(secp256k1.o)
          
          rust-lld: error: duplicate symbol: WASM32_PTR_ALIGN
          >>> defined in /tmp/rustciBlixx/libsecp256k1_sys-b664e02e63969bc9.rlib(precomputed_ecmult.o)
          >>> defined in /tmp/rustciBlixx/libsecp256k1_sys-b664e02e63969bc9.rlib(secp256k1.o)
```

</p>
</details>

### CODE REVIEW

**Comment by junderw** [comment]
I also get these warnings:

<details><summary>Expand</summary>
<p>

```
warning: In file included from depend/secp256k1/src/secp256k1.c:16:
warning: depend/secp256k1/src/group_impl.h:122:13: warning: unused function 'rustsecp256k1_v0_5_0_ge_set_all_gej_var' [-Wunused-function]
warning: static void rustsecp256k1_v0_5_0_ge_set_all_gej_var(rustsecp256k1_v0_5_0_ge *r, const rustsecp256k1_v0_5_0_gej *a, size_t len) {
warning:             ^
warning: In file included from depend/secp256k1/src/secp256k1.c:17:
warning: In file included from depend/secp256k1/src/ecmult_impl.h:16:
warning: In file included from depend/secp256k1/src/ecmult.h:12:
warning: depend/secp256k1/src/scratch.h:24:38: warning: unused function 'rustsecp256k1_v0_5_0_scratch_create' [-Wunused-function]
warning: static rustsecp256k1_v0_5_0_scratch* rustsecp256k1_v0_5_0_scratch_create(const rustsecp256k1_v0_5_0_callback* error_callback, size_t max_size);
warning:                                      ^
warning: depend/secp256k1/src/scratch.h:26:13: warning: unused function 'rustsecp256k1_v0_5_0_scratch_destroy' [-Wunused-function]
warning: static void rustsecp256k1_v0_5_0_scratch_destroy(const rustsecp256k1_v0_5_0_callback* error_callback, rustsecp256k1_v0_5_0_scratch* scratch);
warning:             ^
warning: In file included from depend/secp256k1/src/secp256k1.c:17:
warning: depend/secp256k1/src/ecmult_impl.h:795:12: warning: unused function 'rustsecp256k1_v0_5_0_ecmult_multi_var' [-Wunused-function]
warning: static int rustsecp256k1_v0_5_0_ecmult_multi_var(const rustsecp256k1_v0_5_0_callback* error_callback, rustsecp256k1_v0_5_0_scratch *scratch, rustsecp256k1_v0_5_0_gej *r, const rustsecp256k1_v0_5_0_scalar *inp_g_sc, rustsecp256k1_v0_5_0_ecmult_multi_callback cb, void *cbdata, size_t n) {
warning:            ^
warning: depend/secp256k1/src/ecmult_impl.h:379:12: warning: unused function 'rustsecp256k1_v0_5_0_ecmult_strauss_batch_single' [-Wunused-function]
warning: static int rustsecp256k1_v0_5_0_ecmult_strauss_batch_single(const rustsecp256k1_v0_5_0_callback* error_callback, rustsecp256k1_v0_5_0_scratch *scratch, rustsecp256k1_v0_5_0_gej *r, const rustsecp256k1_v0_5_0_scalar *inp_g_sc, rustsecp256k1_v0_5_0_ecmult_multi_callback cb, void *cbdata, size_t n) {
warning:            ^
warning: depend/secp256k1/src/ecmult_impl.h:621:15: warning: unused function 'rustsecp256k1_v0_5_0_pippenger_scratch_size' [-Wunused-function]
warning: static size_t rustsecp256k1_v0_5_0_pippenger_scratch_size(size_t n_points, int bucket_window) {
warning:               ^
warning: depend/secp256k1/src/ecmult_impl.h:704:12: warning: unused function 'rustsecp256k1_v0_5_0_ecmult_pippenger_batch_single' [-Wunused-function]
warning: static int rustsecp256k1_v0_5_0_ecmult_pippenger_batch_single(const rustsecp256k1_v0_5_0_callback* error_callback, rustsecp256k1_v0_5_0_scratch *scratch, rustsecp256k1_v0_5_0_gej *r, const rustsecp256k1_v0_5_0_scalar *inp_g_sc, rustsecp256k1_v0_5_0_ecmult_multi_callback cb, void *cbdata, size_t n) {
warning:            ^
warning: 7 warnings generated.
```

</p>
</details>

**Comment by junderw** [comment]
The precompute was turning my 67kB WASM binary to 1200kB even with stripping and wasm-opt etc.

I added a feature to remove the precompute.

File: secp256k1-sys/Cargo.toml [hunk 1]
```diff
@@ -29,5 +29,6 @@ libc = "0.2"
 default = ["std"]
 recovery = []
 lowmemory = []
+noprecompute = []
```
Comment (thomaseizinger) [inline secp256k1-sys/Cargo.toml:32]: Should we define this as the inverse and say "precompute"? Feature flags in Rust are additive so they should _add_ functionality.

If someone relies on this and adds another dependency to their graph that enables this, their code will break.

**Review by thomaseizinger — APPROVED** [review]

File: secp256k1-sys/Cargo.toml (hunk 1 above)
Comment (junderw) [reply secp256k1-sys/Cargo.toml]: Done. Thanks.

**Comment by junderw** [comment]
I'm stumped...

**Comment by junderw** [comment]
I was able to get the tests to run, but it seems like a lot of pubkey related (ecmult related) tests are giving weird results...

Not sure exactly what could be causing it, but it's something to do with the diff here:

https://github.com/rust-bitcoin/rust-secp256k1/compare/ce255fdfe1423f85a72d593d721420704b5a80df...8294ea3f50f6de88f88b848c01eb60cab2a73241

(0.21.3 <-> 0.22.1)

**Comment by tcharding** [comment]
Thanks for having a go!

**Pull request closed without merging**